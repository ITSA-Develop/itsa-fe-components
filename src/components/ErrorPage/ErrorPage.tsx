import { Button } from '@/components/Button/Button';
import { Typography } from 'antd';
export interface ErrorPageProps {
	error: string;
	message: string;
	handleClick: () => void;
}

const { Title } = Typography;

export const ErrorPage = ({ error, message, handleClick }: ErrorPageProps) => {
	return (
		<div className="flex flex-col justify-center items-center h-screen bg-white-100">
			<div className="text-center">
				<Title>{error}</Title>
				<p className="text-lg">{message}</p>
			</div>
			<div className="flex justify-center mt-8">
				<Button onClick={handleClick}>Ir al Inicio</Button>
			</div>
		</div>
	);
};
