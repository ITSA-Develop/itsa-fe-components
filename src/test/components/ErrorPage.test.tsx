import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { ErrorPage } from '../../components/ErrorPage/ErrorPage';

describe('ErrorPage component', () => {
	it('renders error and message texts', () => {
		const { container } = render(<ErrorPage error="404" message="Página no encontrada" handleClick={() => {}} />);

		expect(screen.getByText('404')).toBeInTheDocument();
		expect(screen.getByText('Página no encontrada')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('calls handleClick when button is clicked', () => {
		const handleClick = vi.fn();
		render(<ErrorPage error="500" message="Error interno" handleClick={handleClick} />);

		const button = screen.getByRole('button', { name: /Ir al Inicio/i });
		fireEvent.click(button);

		expect(handleClick).toHaveBeenCalledTimes(1);
	});
});
