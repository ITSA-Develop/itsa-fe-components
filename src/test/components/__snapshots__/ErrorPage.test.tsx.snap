// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage component > renders error and message texts 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center h-screen bg-white-100"
  >
    <div
      class="text-center"
    >
      <h1
        class="ant-typography css-dev-only-do-not-override-l9pxc0"
      >
        404
      </h1>
      <p
        class="text-lg"
      >
        PÃ¡gina no encontrada
      </p>
    </div>
    <div
      class="flex justify-center mt-8"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
        type="button"
      >
        <span>
          Ir al Inicio
        </span>
      </button>
    </div>
  </div>
</div>
`;
